docs/source/jwst/datamodels/models.rst

```diff
@@ -213,154 +158,217 @@ Searching a model
     └─filteroffset (dict)
 
+Data Arrays
+-----------
+Most datamodels have one or more data arrays as attributes.
+For most datamodels containing science data, the primary data array is
+accessible through the ``data`` attribute.  Other commonly used data arrays
+are ``dq`` (data quality) and ``err`` (uncertainty/error).
+
+The ``data``, ``err``, ``dq``, etc., attributes of most models are assumed to be
+numpy.ndarray arrays, or at least objects that have some of the attributes
+of these arrays.  ``numpy`` is used explicitly to create these arrays in some
+cases (e.g. when a default value is needed).  The ``data`` and ``err`` arrays
+are a floating point type, and the data quality arrays are an integer type.
+
+Metadata
+--------
+Metadata information associated with a data model is accessed through
+its `meta` member.  For example, to access the date that an
+observation was made::
 
+    print(model.meta.observation.date)
 
-Converting from ``astropy.io.fits``
-===================================
-
-This section describes how to port code that uses ``astropy.io.fits``
-to use `jwst.datamodels`.
-
-.. _datamodels-open:
-
-Opening a file
---------------
-
-Instead of::
-
-    astropy.io.fits.open("myfile.fits")
-
-use::
-
-    from stdatamodels.jwst.datamodels import ImageModel
-    with ImageModel("myfile.fits") as model:
-        ...
-
-In place of `ImageModel`, use the type of data one expects to find in
-the file.  For example, if spectrographic data is expected, use
-`SpecModel`.  If it doesn't matter (perhaps the application is only
-sorting FITS files into categories) use the base class `JwstDataModel`.
-
-An alternative is to use::
-
-    from stdatamodels.jwst import datamodels
-    with datamodels.open("myfile.fits") as model:
-        ...
-
-The `datamodels.open()` method checks if the `DATAMODL` FITS keyword has
-been set (or, for ASDF files, the `meta.model_type attribute`), 
-which records the DataModel that was used to create the file.
-If the keyword is not set, then `datamodels.open()` does its best to
-guess the best DataModel to use and emits a `NoTypeWarning`.
-
-Accessing data
---------------
-
-Data should be accessed through one of the pre-defined data members on
-the model (`data`, `dq`, `err`).  There is no longer a need to hunt
-through the HDU list to find the data.
-
-Instead of::
-
-    hdulist['SCI'].data
-
-use::
-
-    model.data
-
-Accessing keywords
-------------------
-
-The data model hides direct access to FITS header keywords.  Instead,
-use the :ref:`metadata` tree.
-
-There is a convenience method, `find_fits_keyword` to find where a
-FITS keyword is used in the metadata tree::
+Metadata values are automatically type-checked against the schema when
+they are set. Therefore, setting a keyword which expects a number to a
+string will raise an exception::
+
+    >>> from stdatamodels.jwst.datamodels import ImageModel
+    >>> model = ImageModel()
+    >>> model.meta.target.ra = "foo"    # doctest: +SKIP
```

Comment:
If you want to make sure exception is still thrown, you can try this instead of skipping it: https://docs.python.org/3/library/doctest.html#doctest.IGNORE_EXCEPTION_DETAIL